File: bug-tracker/src/main/java/rrpay/bug/model/Bug.java
package rrpay.bug.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import rrpay.bug.model.enums.BugStatus;
import rrpay.bug.model.enums.BugType;
import rrpay.bug.model.enums.Priority;

import java.nio.file.Path;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
@Document(collection="bug")
public class Bug {
  @Id
  private ObjectId id;
  private String code;
  private String summary; //user input
  private String description; //user input
  private Priority priority = Priority.NONE;
  private Project project;
  private User reporter;
  private User assignee;
  private LocalDateTime dateCreated;
  private LocalDateTime dateResolved;
  private LocalDateTime dateClosed;
  private BugType type;
  private BugStatus status = BugStatus.OPEN;
  private List<Comment> comments;
  private List<String> workHistory;
  private Set<Path> attachments;

}

File: bug-tracker/src/main/java/rrpay/bug/kt/model/Bug.kt
package rrpay.bug.kt.model

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import rrpay.bug.model.Comment
import rrpay.bug.model.Project
import rrpay.bug.model.User
import rrpay.bug.model.kt.enums.BugStatus
import rrpay.bug.model.kt.enums.BugType
import rrpay.bug.model.kt.enums.Priority
import java.nio.file.Path
import java.time.LocalDateTime

@Document(collection = "bug")
data class Bug(
    var summary: String,
    var description: String,
    var priority: Priority = Priority.NONE,
    var status: BugStatus = BugStatus.NEW,
) {
    constructor(
        summary: String,
        description: String,
        priority: Priority = Priority.NONE,
        status: BugStatus = BugStatus.NEW, type: BugType = BugType.SOFTWARE_BUG
    ) : this(
        summary,
        description,
        priority,
        status
    ) {
        this.type = BugType.SOFTWARE_BUG
    }

    @Id
    var id: ObjectId? = null
    var code: String = ""
    var project: Project? = null
    var reporter: User? = null
    var assignee: User? = null
    var dateCreated: LocalDateTime? = null
    var dateResolved: LocalDateTime? = null
    var dateClosed: LocalDateTime? = null
    var comments: List<Comment>? = null
    var workHistory: List<String>? = null
    var attachments: Set<Path>? = null
    var type: BugType? = null

}


